lib/1.4/dml-builtins.dml

```diff
@@ -36,18 +35,23 @@ constant unmapped_offset = 0xffff_ffff_ffff_ffff;
 
 param NULL auto;
 
-// Handy macros
+// Handy macros and types
+extern int strcmp(const char *__s1, const char *__s2);
+extern void *memset(void *__s, int __c, size_t __n);
+extern typedef ulong uintptr_t;
```

Comment:
Redundant: ulong is provided by types.py (and incorrect IIRC: uintptr_t is `unsigned long long` on one of the platforms)

Reply:
Wait.

_Oh crap._ We need to build-in `uintptr_t`.

Redundant is a **horribly** incorrect word for it, by the by!

Reply:
Looking at dml-api, `unsigned long long` gets translated to `uint64`. So we don't need to have it built-in, we can just use `uint64`.

Reply:
... Or `uint64_t`? Hm. `uint64_t` is likely the better choice, because IIRC the same ulong vs. unsigned long long dichotomy applies to it as it does to `uintptr_t`. That is, they are identical on both hosts.

Reply:
Picked `uint64_t`